### PR TITLE
Pwa 755

### PR DESCRIPTION
**Pull request by alexbridge**

### CODE REVIEW

File: frontend/pages/MyProfile/style.js [hunk 1]
```diff
@@ -6,8 +6,6 @@ const isGmd = themeName.includes('gmd');
 const { variables, colors } = themeConfig;
 
 const container = css({
-  paddingLeft: variables.gap.big,
```
Comment (reneeichhorn) [inline frontend/pages/MyProfile/style.js:9]: This causes the headline to have no padding at all.

File: frontend/subscriptions/ui.js [hunk 2]
```diff
@@ -34,13 +35,15 @@ export default (subscribe) => {
...
+      routeDidEnter(USER_ADDRESS_PATH),
+      routeDidEnter(USER_PASSWORD_PATH)
     );
   const fullPageViewLeave$ = routeDidLeave(USER_REGISTER_PATH)
     .merge(
       routeDidLeave(USER_PROFILE_PATH),
       routeDidLeave(USER_ADDRESS_BOOK_PATH),
-      routeDidLeave(USER_ADDRESS_PATH)
+      routeDidLeave(USER_ADDRESS_PATH),
+      routeDidEnter(USER_PASSWORD_PATH)
```
Comment (reneeichhorn) [inline frontend/subscriptions/ui.js]: typo: should be `routeDidLeave` ;)

File: frontend/subscriptions/user.js [hunk 3]
```diff
@@ -60,4 +62,19 @@ export default (subscribe) => {
 
     dispatch(createToast({ message: 'profile.updated' }));
   });
+
+  // Modal message when profile or password update failed
+  subscribe(userUpdateFailed$, ({ dispatch, action }) => {
```
Comment (reneeichhorn) [inline frontend/subscriptions/user.js:74]: I think we should redirect the user to the profile overview page after a successful password change

File: extension/user/update/updatePassword.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+const InternalError = require('./../../common/Error/InternalError')
+const InvalidCredentialsError = require('./../../common/Error/InvalidCredentialsError')
+const ValidationError = require('./../../common/Error/ValidationError')
+const Password = require('./../Password')
```
Comment (reneeichhorn) [inline extension/user/update/updatePassword.js:4]: I just realized that we are using sha256 for password encryption, shouldn't we change that to bcrypt? Or is this pipeline implementation only intended for dev only anyway? What about the legacy user?

File: frontend/components/ChangePasswordForm/index.jsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            onChange={this.handleRepeatPassword}
+            errorText={this.state.errors.repeatPassword}
+          />
+
+          {!isIos &&
+            <Grid className={styles.buttons}>
+              <Grid.Item grow={1} />
+              <Grid.Item grow={0} className={styles.cancelItem}>
+                <Link
+                  href={USER_PROFILE_PATH}
```
Comment (reneeichhorn) [inline frontend/components/ChangePasswordForm/index.jsx]: I don't think that we should use the Link component. That would push an history entry to history stack and when using the back button you would go back to the password change.
The cancel should trigger a history goBack action.

**Review by reneeichhorn — CHANGES_REQUESTED** [review]

File: extension/user/update/updatePassword.js (hunk 4 above)
Comment (alexbridge) [reply extension/user/update/updatePassword.js:4]: The goal is to delegate this functionality to external apis. 
We just give a demo or simple user implementation here.
After we connect extensions together, we will see how to handle passwords

**Review by reneeichhorn — APPROVED** [review]

**Pull request merged**